c_src/quicer_nif.c

```diff
@@ -513,42 +513,55 @@ resource_listener_down_callback(__unused_parm__ ErlNifEnv *caller_env,
 }
 
 void
-resource_conn_dealloc_callback(__unused_parm__ ErlNifEnv *caller_env,
-                               void *obj)
+resource_conn_dealloc_callback(__unused_parm__ ErlNifEnv *env, void *obj)
 {
   QuicerConnCTX *c_ctx = (QuicerConnCTX *)obj;
+  TP_CB_3(start, c_ctx->Connection, 0);
   enif_demonitor_process(c_ctx->env, c_ctx, &c_ctx->owner_mon);
   AcceptorQueueDestroy(c_ctx->acceptor_queue);
   enif_free_env(c_ctx->env);
   enif_mutex_destroy(c_ctx->lock);
   CXPLAT_FREE(c_ctx->TlsSecrets, QUICER_TLS_SECRETS);
   CXPLAT_FREE(c_ctx->ssl_keylogfile, QUICER_TRACE);
   AcceptorDestroy(c_ctx->owner);
+  TP_CB_3(end, c_ctx->Connection, 0);
 }
 
 void
-resource_conn_down_callback(__unused_parm__ ErlNifEnv *caller_env,
-                            __unused_parm__ void *obj,
+resource_conn_down_callback(__unused_parm__ ErlNifEnv *env,
+                            void *ctx,
                             __unused_parm__ ErlNifPid *pid,
                             __unused_parm__ ErlNifMonitor *mon)
 {
-  // todo
+  QuicerConnCTX *c_ctx = ctx;
+  if (!ctx)
+    {
+      return;
+    }
+  else
+    {
+      TP_CB_3(start, c_ctx->Connection, (uint64_t)ctx);
+      MsQuic->ConnectionShutdown(
```

Comment:
ConnectionShutdown returns void, no way to check if the shutdown is success or not.